src/core/Exception.pm6

```diff
@@ -171,7 +171,11 @@ my class X::Method::NotFound is Exception {
         }
 
         if nqp::can($!invocant.HOW, 'methods') {
-            for $!invocant.^methods(:all)>>.name -> $method_name {
+            my @invocant_methods = $!invocant.^methods(:local)>>.name;
```

Comment:
This could be `my @invocant_methods = $!invocant.^methods(:local).grep(*.^name eq 'Submethod')>>.name;`, right? That way the list to check inside the loop is smaller.

Reply:
Whoops, I didn't realize there was a comment here.
Yeah, I can move it there, but it's a bit of a pessimization as well, because then we calculate the list of submethods even if we're never to suggest them.

Reply:
No, I'm fine with it.